azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodBuilder.java

```diff
@@ -0,0 +1,189 @@
+package azkaban.container.models;
+
+import io.kubernetes.client.custom.Quantity;
+import io.kubernetes.client.openapi.models.V1Container;
+import io.kubernetes.client.openapi.models.V1ContainerBuilder;
+import io.kubernetes.client.openapi.models.V1EnvVar;
+import io.kubernetes.client.openapi.models.V1EnvVarBuilder;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1ObjectMetaBuilder;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodBuilder;
+import io.kubernetes.client.openapi.models.V1ResourceRequirements;
+import io.kubernetes.client.openapi.models.V1ResourceRequirementsBuilder;
+import io.kubernetes.client.openapi.models.V1Volume;
+import io.kubernetes.client.openapi.models.V1VolumeBuilder;
+import io.kubernetes.client.openapi.models.V1VolumeMount;
+import io.kubernetes.client.openapi.models.V1VolumeMountBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A wrapper Kubernetes Pod builder which internally utilizes
+ * @see io.kubernetes.client.openapi.models to create kubernetes "Pod" resource
+ *
+ * This class provides custom builder methods to have tight coupling of
+ * volumes and init containers (associated with job-types) with application-container
+ * enabling better type safety and compact implementation
+ *
+ * The Pod-spec created using this builder creates one application-container
+ * and multiple init containers corresponding to each jobType
+ */
+public class AzKubernetesV1PodBuilder {
```

Comment:
We will need similar type of builder for service as well. What do you think?

Reply:
It is not necessary to have a custom builder for each "kind". In the case of Pod, we wanted a tight coupling of init contaienrs, init container volume mounts, app container volume mounts, with the app/flow container, which will be error prone if we use builders separately.
However, if such a use case arise again for any other "kind" of resource, then we can have a custom builder for each kind.

Reply:
There is requirement to create service. So we will need custom builder for it now itself.